app/views/includes/forms/inputRadioGroupStylingWithH1Legend.scala.html

```diff
@@ -0,0 +1,77 @@
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*@
+
+@(field: Field,
+radioOptions: Seq[(String, String)],
+legend: String,
+legendHint: Option[String],
+mainHeadingText: String,
+preHeadingText: String,
+preHeadingAccessibleAnnouncement: String,
+args: (Symbol, Any)*)(implicit messages: Messages)
+
+@import play.api.i18n._
+@import views.html.helper._
+
+@elements = @{new FieldElements(field.id, field, null, args.toMap, messages) }
+@groupDivClass = {@elements.args.get('_groupDivClass)@if(elements.hasErrors){ form-group-error}}
+@fieldsetClass = {@elements.args.get('_groupClass)}
+@labelAfter = @{ elements.args.get('_labelAfter).getOrElse(false).asInstanceOf[Boolean] }
+
+
+@preHeadingHidden = @{
+    if(preHeadingAccessibleAnnouncement=="") Messages("tai.ptaHeader.accessible.preHeading") else preHeadingAccessibleAnnouncement
+}
+
+
+<div class="@groupDivClass">
+
+    <fieldset @if(fieldsetClass != ""){class="@fieldsetClass"} @if(elements.args.get('_fieldsetAttributes).isDefined) {@elements.args.get('_fieldsetAttributes)}  @elements.errors.map{error => aria-describedby="error-message-@{field.id}-input"}>
+
+    <legend @if(elements.args.get('_legendClass).isDefined) {class="@elements.args.get('_legendClass)"}>
+    <div class="page-heading soft--top">
+    <h1 class="heading-xlarge heading-mobile-responsive break-word">@Html(mainHeadingText)</h1>
+    <p class="heading-secondary break-word"><span class="visuallyhidden">@preHeadingHidden </span>@Html(preHeadingText)</p>
+    </div>
+        @if(legendHint.isDefined) {<span id="radioGroupLegendHint">@legendHint.get</span>}
+    </legend>
+
+    @elements.errors.map{error => <p class="error-message" id="error-message-@{field.id}-input"><span class="visuallyhidden">@Messages("tai.error.message") </span>@Messages(error.toString)</p>}
+
+    @for(((value, label), optionIdx) <- radioOptions zip (Stream from 1)) {
+
+    @defining(s"${elements.field.name}-${value.toLowerCase.replace(" ","_")}")  { inputId =>
+    <div class="multiple-choice" @elements.args.get( Symbol(s"_radioOption${optionIdx}WrapAttribute") )>
+    <input  type="radio"
+            id="@inputId"
+            name="@elements.field.name"
+            value="@value"
+            @elements.args.get('_inputClass).map{inputClass => class="@inputClass"}
+    @if(elements.args.contains(Symbol(s"_radioOption${optionIdx}Attributes")) ){ @elements.args.get( Symbol(s"_radioOption${optionIdx}Attributes") )}
+    @field.value.filter( _ == value).map{_ => checked="checked"}/>
```

Comment:
nitpick, i think this could be more concise with something like: `collect { case `value` => checked="checked" }` or 
`collect { case x if x == value => checked="checked" }`

Reply:
thats fine @sergi-hmrc  - however just flagging this for information......we have taken a copy of the existing  component code inputRadioGroupStyling and the only difference is that inputRadioGroupStylingWithH1Legend has the H1 inside legend instead.  Making other improvement changes elsewhere here result in the 2 similar code pieces diverging.

Reply:
no worries! it was just a nitpick I saw :P